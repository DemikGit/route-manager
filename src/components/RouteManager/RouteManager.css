.route-manager__container {
  display: flex;
  width: 100%;
  height: 100%;
  max-width: 1024px;
  margin-left: 20px;
  margin-right: 20px;
  justify-content: space-between;
}

.route-manager__controls {
  display: flex;
  flex-basis: 50%;
  flex-direction: column;
  justify-content: flex-start;
}

.route-manager__map {
  display: block;
  width: 50%;
  min-width: 300px;
  height: 100%;
}
