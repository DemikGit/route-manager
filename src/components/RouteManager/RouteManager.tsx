import * as React from 'react';
import { Component } from 'react';
import { BaseInput } from '../BaseInput/BaseInput';
import './RouteManager.css'

import { GoogleMap } from "react-google-maps"
import { RMRoute } from '../../classes/RMRoute';
import { MapPreview } from '../MapPreview/MapPreview';
import { MarkerList } from '../MarkersList/MarkerList';

export interface IRouteManagerProps {
  text: string,
}
export interface IRouteManagerState {
  apiParams: {
    key: string,
    libs: string,
    version: string,
  },
  route: RMRoute,
}

export class RouteManager extends Component<IRouteManagerProps, IRouteManagerState> {
  public state = {

    apiParams: {
      key: 'AIzaSyDSnQhXn5WfyBi68hCgx5qA4pmApKkN8Jc',
      libs: 'geometry,drawing,places',
      version: '3.exp',
    },

    route: new RMRoute(),
  }

  private mapRef: React.RefObject<GoogleMap> = React.createRef();

  public componentDidMount() {
    const { route } = this.state;
    route.mapRef = this.mapRef;
    route.updateComponent = () => this.forceUpdate();
  }

  public render() {
    const { route } = this.state;
    const { key, libs, version } = this.state.apiParams;

    const url = `https://maps.googleapis.com/maps/api/js?key=${
      key }&v=${ version }&libraries=${ libs }`;

    return (
      <div className="route-manager__container">
        <div className="route-manager__controls">
          <BaseInput text="kek" onAddPoint={ route.onAddPoint } />
          <MarkerList routeMarkers={ route.markers } onPointDelete={ route.onPointDelete } />
        </div>
        <div className="route-manager__map">
          <MapPreview
            googleMapURL={ url }
            defaultCenter={{
              lat: 37.666030368995685,
              lng: -116.02545749206541,
            }}
            defaultZoom={ 15 }
            loadingElement={ <div style={{ height: '100%' }} /> }
            containerElement={
              <div className="route-manager__map__wrapper"/>
            }
            mapElement={ <div style={{ height: '100%' }} /> }
            mountMap={ this.mapRef }
          >
            { route.render() }
          </MapPreview>
        </div>
      </div>
    );
  }
}
