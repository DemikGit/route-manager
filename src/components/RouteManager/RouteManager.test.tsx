import { shallow, mount } from 'enzyme';
import 'jest-enzyme';
import * as React from 'react';
import { RouteManager } from './RouteManager';

it('renders without crashing', () => {
  const wrapper = shallow(<RouteManager />);
  expect(wrapper).toMatchSnapshot();
});


it('call createNewPoint if Enter key pressed', () => {
  const wrapper = shallow(<RouteManager />);
  const createNewPointMock = jest.fn();
  const mockEvent = {
    nativeEvent: {
      key: 'Enter',
    }
  }
  const instance = wrapper.instance() as RouteManager;

  instance.createNewPoint = createNewPointMock
  instance.onKeyUp(mockEvent as React.KeyboardEvent<HTMLInputElement>) ;

  expect(createNewPointMock).toBeCalled();
  }
);

it('call addPoint if input value exist', () => {
  const wrapper = shallow(<RouteManager />);
  const addPointMock = jest.fn();
  const inputMock = (global.document as Document).createElement('input');
  const mockEvent = {
    nativeEvent: {
      key: 'Enter',
      target: inputMock,
    },
  }
  inputMock.value = 'data';
  const instance = wrapper.instance() as RouteManager;
  instance.addPoint = addPointMock;

  instance.onKeyUp(mockEvent);

  expect(addPointMock).toBeCalled();
  }
);
