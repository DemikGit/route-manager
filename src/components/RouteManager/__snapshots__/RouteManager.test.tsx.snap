// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="route-manager__container"
>
  <div
    className="route-manager__controls"
  >
    <BaseInput
      onKeyUp={[Function]}
      placeholder="Enter name for new routes point..."
    />
    <MarkerList
      onPointDelete={[Function]}
      routeMarkers={Array []}
      swapPoints={[Function]}
    />
  </div>
  <div
    className="route-manager__map"
  >
    <withScriptjs(withGoogleMap(MapPreviewComponent))
      containerElement={
        <div
          className="route-manager__map__wrapper"
        />
      }
      defaultCenter={
        Object {
          "lat": 37.666030368995685,
          "lng": -116.02545749206541,
        }
      }
      defaultZoom={15}
      googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyDSnQhXn5WfyBi68hCgx5qA4pmApKkN8Jc&v=3.exp&libraries=geometry,drawing,places"
      loadingElement={
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      }
      mapElement={
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      }
      mountMap={
        Object {
          "current": null,
        }
      }
    />
  </div>
</div>
`;
